x/distribution/spec/03_end_block.md

```diff
@@ -30,4 +30,7 @@ func AllocateTokens(feesCollected sdk.Coins, feePool FeePool, proposer Validator
 
      SetValidatorDistribution(proposer)
      SetFeePool(feePool)
+
+// Thoughts: I don't understand enough for this distribution logic on endblocker, how it's handling all delegator / validator commissions.
+// Distribution module is saving all the balance by account?
```

Comment:
It keeps track of the number of "delegation shares" that each delegator has.  The number of delegation shares only changes on an increase or decrease in delegation amount.

Reply:
@ValarDragon do you have the PDF that explains f1?

Reply:
https://github.com/cosmos/cosmos-sdk/blob/master/docs/spec/_proposals/f1-fee-distribution/f1_fee_distr.pdf